### DISCUSSION TITLE
翻訳の方針

### DISCUSSION THREAD

**Discussion by krisfail** [Katsuyuki-Karasawa/stable-diffusion-webui-localization-ja_JP#1] [General]
バラバラな方針で翻訳してるとちぐはぐになりそうなので、できるだけ統一した方針を共有しておきたいです。
私が思っていることをとりあえず書いておきます。

- 単なる和訳ではなく翻訳を心がける
  - 日本人に通じなさそうな表現は思い切って意訳します
- 英語と日本語で情報量の多寡があることを考慮する
  - 何でもかんでも日本語訳(特に設定項目の直訳)してしまうと、逆にユーザーが調べにくくなるという弊害がありそう
     - 英語圏の設定を流用するのに設定がどれに該当するかわからないというのはかなり面倒だと思う
  - 特殊な技術用語はできるだけ英語併記したい
- 個別の環境依存なファイル名が紛れ込まないよう注意する
  - vae,ckptなど
- 約物(`、。「」`など)の使い方
  - ` 、。`はUIに直接出てくる文には使わないのが普通？
    - 英語の方も`.`があったりなかったりなので適当でもよさそう
  - 英語の`;`,`:`,`,`,`-`をどう訳出するか 
    - とりあえず`：`や`―`で置き換える
- これは一般向けソフトウェアではないのだから、親しみやすさより機能性を優先する
   - 漢語を活用する:`add`は`加える`と訳出するより`追加`などのほうが引き締まってよい
   - カタカナに変えるだけの訳は親しみやすさ以外に意味がないのでやめたほうが良さそう

**Comment by krisfail** [comment]
- ベースとなるファイル
   - 最新版のWebUIに公式IndexにあるExtensionをすべて導入した状態で、テンプレートを生成させる
     - Localizationが適用されている場合、それが反映されたものが出力される

**Reply by L4Ph** [reply]
これが良い気がします
しいていうのであれば
- Extensionsが新しく追加されるタイミングでdiffがズレる

などでしょうか?
現状は考えなくてよいと思うので、後回しでよさそうですが

**Reply by L4Ph** [reply]
一度(localizationsを除く)全拡張機能を有効化した上で生成してみました
`ddetailer`が動かずエラー吐いたので、これだけ翻訳に含まれていないかもしれないので一度確認するためにgistに上げました
https://gist.github.com/Katsuyuki-Karasawa/3672cd073d20175473b2aa7d14e4e648

**Reply by krisfail** [reply]
余計なものがまじりすぎててちょっと使いにくいですね…
こちらでもテンプレートを出させてみます

**Reply by krisfail** [reply]
あまりにも多すぎるので、とりあえず拡張機能はAUTOMAITC1111製のものと、`NAIConvert`のような機能の少ないものに絞って始めるのがよいかと。
WebUI本体の翻訳を改善し、sd2.0に備えるのが最優先ですかね

**Reply by L4Ph** [reply]
そうしたらとりあえずATOMATIC1111製のものから手をつけるのがよさそうですね。
ノーマルのものがほしいので、とりあえず初回起動して直後(なんらExtensionsをDLしてない状態)のjsonを配置します

**Reply by L4Ph** [reply]
https://github.com/Katsuyuki-Karasawa/stable-diffusion-webui-localization-ja_JP/commit/ab8ff28eb6d74e5203b7b7262b9b24e4cab262cb
https://github.com/Katsuyuki-Karasawa/stable-diffusion-webui-localization-ja_JP/commit/472959a0f2a5c0463ef0a06d9aa676e10529c47f
こちらを追加しました。
Extensionsはユーザー単位で抽出しようと思います。

**Reply by krisfail** [reply]
とりあえず自分は現在翻訳ファイルを利用している人向けに既存の翻訳を改善する方向の作業を当分継続します。
唐澤さんが手を加えた部分に明らかな誤訳が多いように思います。自分の不手際から始まってメンテナンスさせているのに文句をつけるのは申し訳ないんですが、助詞の扱いが怪しかったり、Extensionタブの`available`を有効化と訳すのは流石に問題かと…

**Reply by L4Ph** [reply]
こちらこそ、わざわざ切り離していただいたのにも関わらず申し訳ありません。
取り急ぎで行っていたため、かなりミスの多い翻訳となっていました(今思うとそのままpushするべきではなかったです。)
私も当面は現行の翻訳を修正する方向で行っていこうと考えていましたので、助かります。
たびたびご迷惑をおかけし、申し訳ありません。
もしあまりにも目につくようであればindexをyuuki76さんのほうに戻していただき、readmeなど必要なもののみ取り込んでいただいて全くかまいませんので、おっしゃっていただければと思います。

**Reply by krisfail** [reply]
ここで続けることに特に異論はないです。(学業のほうが忙しく、度々放置することになりそうですし)
というか、僕の翻訳も相当雑なものですし、こういうものは個人の言語スキルというより人海戦術で解決するものでしょう。
いずれにせよ、人員不足がかなり深刻なので、早めに翻訳プラットフォームを導入してgitやjsonに疎い人でも翻訳に参加できるようにしたほうが良いと思います
